modules/dists/ReplicatedDist.chpl

```diff
@@ -232,18 +232,16 @@ proc ReplicatedDist.dsiSupportsPrivatization() param return true;
 proc ReplicatedDist.dsiGetPrivatizeData() {
   if traceReplicatedDist then writeln("ReplicatedDist.dsiGetPrivatizeData");
 
-  // TODO: return the targetLocales array by value,
-  // to reduce communication needed in dsiPrivatize()
-  // perhaps by wrapping it in a class (or tuple?).
-  return targetLocales;
+  // TODO: Returning 'targetLocales' here results in a memory leak. Why?
```

Comment:
ReplicatedDist is AFAIK the only one in which targetLocales is not of defined type (rectangular, rank based on distribution). Instead it tries to work with associative array targetLocales, which is unusual.